### PR TITLE
Fixed issues in the dashboard designer and view mode

### PR DESCRIPTION
**Pull request by lasanthaS**

### CODE REVIEW

File: apps/portal/controllers/dashboards.jag [hunk 1]
```diff
@@ -46,6 +46,41 @@ var fresh;
             response.sendError(404, 'requested dashboard cannot be found');
             return;
         }
+        
+        // Verify the original dashboard when returning a custom dashboard
+        if (dashboard.isUserCustom) {
+
```
Comment (TanyaM) [inline apps/portal/controllers/dashboards.jag]: Remove new line

File: apps/portal/controllers/dashboards.jag [hunk 2]
```diff
@@ -46,6 +46,41 @@ var fresh;
...
+            // To view personalized dashboard, the user should have logged in, in the same domain and should have either editor or viewer permissions
+            if (user && user.domain == (urlDomain || superDomain) && 
+                (utils.allowed(user.roles, originalDashboard.permissions.editors) || utils.allowed(user.roles, originalDashboard.permissions.viewers))) {
+                authorized = true;                
+            }
+            
+            if (! authorized) {
+                // If there is no anon view defined send the unauthorized error, else return the anon view
+                if (!originalDashboard.isanon) {
+                    response.sendError(401, 'access to this dashboard is denied');
```
Comment (TanyaM) [inline apps/portal/controllers/dashboards.jag]: Make error message more descriptive (mention to which dashboard)

File: apps/portal/controllers/dashboards.jag [hunk 3]
```diff
@@ -46,6 +46,41 @@ var fresh;
...
+            
+            // Get the original dashboard
+            var originalDashboard;
+            utils.sandbox(context, function() {
+                originalDashboard = dashboards.findOne(dashboardId, true);
+            });
+        
+            // If the original dashboard is deleted, returns 404
+            if (! originalDashboard) {
+                response.sendError(404, 'requested dashboard cannot be found');
```
Comment (TanyaM) [inline apps/portal/controllers/dashboards.jag]: Make error more meaningful

File: apps/portal/controllers/dashboards.jag (hunk 1 above)
Comment (lasanthaS) [inline apps/portal/controllers/dashboards.jag]: This is fixed

File: apps/portal/controllers/dashboards.jag (hunk 2 above)
Comment (lasanthaS) [inline apps/portal/controllers/dashboards.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/controllers/dashboards.jag (hunk 3 above)
Comment (lasanthaS) [inline apps/portal/controllers/dashboards.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

**Pull request merged**